guides/quickstart.tt

```diff
@@ -0,0 +1,187 @@
+[% WRAPPER layout.tt hideTitle=1 title="Quickstart guide" menu='nixos' %]
+
+<h1>Quickstart guide</h1>
+
+<p class="lead">This quickstart guide will get you started with the Nix package manager.</p>
+
+<p>You will learn how to…</p>
+
+<ul>
+  <li>Install Nix</li>
+  <li>Create your first Nix shell as a working environment</li>
+  <li>Setup a development environment for a web app</li>
+  <li>Build a Docker image to deploy your web app</li>
+</ul>
+
+
+<h2>Install Nix</h2>
+
+<p>While NixOS is a Linux distribution based on Nix, you can install Nix on other Linux distributions, MacOS and Windows via WSL using the install script from our website:</p>
+
+<pre class="code">
+curl -L https://nixos.org/nix/install | sh
+</pre>
+
+<p>(For security, you may want to <a href="[%root%]download.html#nix-verify-installation">verify the script</a> using our GPG signatures.)</p>
+
+<p>Check that the installation was successful by running</p>
+
+<pre class="code">
+nix-channel --list
+</pre>
+
+<p>This command displays the package distribution channel used by Nix. By default, this is <code>https://nixos.org/channels/nixpkgs-unstable</code></p>
+
+
+<h2>Your first shell</h2>
+
+<p>Nix can create reproducible environments given a declarative configuration called a Nix expression. Reproducible means you can share the configuration with others and guarantee that they are using the same software as you.</p>
+
+<p>To get started, make a new folder and create a file called <code>shell.nix</code> with the following contents:</p>
+
+<pre class="code">
+{ pkgs ? import &lt;nixpkgs&gt; {} }:
+
+pkgs.mkShell {
```

Comment:
`mkShell` is a development shell and will run setup hooks which, depending on your use case, you want or not. Because of that reason it should be avoided unless necessary. This is essentially also the `nix run` versus `nix-shell` thing. If you just want an environment with several packages, use a plain list.

Reply:
Okay, I don't think I can follow you entirely. I interpret "use a plain list" as this:

```
{ pkgs ? import <nixpkgs> {} }:

[
  pkgs.which
  pkgs.htop
]
```

But Nix does not accept that code as a `shell.nix`. Clearly I am missing something…

Reply:
Sorry, you are right. It won't work because `nix-shell` needs to set up an environment. That is unfortunate so I suppose what you have can stay then as the preferred solution. How happy I am with `nix run` :)